### PR TITLE
NMS-8947: Close HttpConnection after usage

### PR DESCRIPTION
**Pull request by mvrueden**
* JIRA: https://issues.opennms.org/browse/NMS-8947

### CODE REVIEW

**Comment by j-white** [comment]
The `fieldInfoCache` also invokes `getConnection()`, but that one doesn't appear to be closed.

**Comment by j-white** [comment]
:+1: from me, waiting on Bamboo

File: features/ticketing/jira-integration/pom.xml [hunk 1]
```diff
@@ -70,5 +70,23 @@
...
+            <type>pom</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish.jersey.core</groupId>
+            <artifactId>jersey-common</artifactId>
+            <version>2.4.1</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Comment (soleger) [inline features/ticketing/jira-integration/pom.xml]: This dep is a dupe.

File: features/ticketing/jira-integration/pom.xml [hunk 2]
```diff
@@ -70,5 +70,23 @@
             <artifactId>junit</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.opennms.dependencies</groupId>
+            <artifactId>spring-dependencies</artifactId>
+            <type>pom</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Comment (soleger) [inline features/ticketing/jira-integration/pom.xml]: I don't see why this jersey dep was added.

**Comment by mvrueden** [comment]
The Jersey and Spring dependencies were added, as they are required by the `JiraTicketerPlugin` to work. Usually the default implementation from Atlassian is providing those dependencies. However we repackage it to make it a osgi bundle. While doing that we do not include spring and jersey. In order to make the tests work, we need those dependencies, otherwise `ClassNotFoundExceptions` are thrown all over the place.

**Pull request merged**